addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionSecurity.java

```diff
@@ -0,0 +1,118 @@
+import org.nuxeo.retention.RetentionConstants;
+import org.nuxeo.retention.adapters.RetentionRule;
+import org.nuxeo.runtime.test.runner.Deploy;
+
+/**
+ * @since 11.1
+ */
+@Deploy("org.nuxeo.retention.core:OSGI-INF/retention-security.xml")
+public class TestRetentionSecurity extends RetentionTestCase {
+
+    @Test
+    public void shouldBeAuthorizedToManageLegalHold() throws LoginException {
+        try (CloseableCoreSession userSession = CoreInstance.openCoreSession(session.getRepositoryName(), "user")) {
+            service.attachRule(file, createManualImmediateRuleMillis(100L), userSession);
+            fail("Sould not be abe to attach rule");
+        } catch (NuxeoException e) {
+            // Expected
+            assertEquals(javax.servlet.http.HttpServletResponse.SC_FORBIDDEN, e.getStatusCode());
+            assertFalse(session.isRecord(file.getRef()));
+            assertFalse(session.isUnderRetentionOrLegalHold(file.getRef()));
+            assertFalse(session.getDocument(file.getRef()).hasFacet(RetentionConstants.RECORD_FACET));
+        }
+    }
+
+    @Test
+    public void shouldNotBeAuthorizedToAttachRule() throws LoginException {
+        try (CloseableCoreSession userSession = CoreInstance.openCoreSession(session.getRepositoryName(), "user")) {
+            service.attachRule(file, createManualImmediateRuleMillis(100L), userSession);
+            fail("Sould not be abe to attach rule");
+        } catch (NuxeoException e) {
+            // Expected
+            assertEquals(javax.servlet.http.HttpServletResponse.SC_FORBIDDEN, e.getStatusCode());
+            assertFalse(session.isRecord(file.getRef()));
+            assertFalse(session.isUnderRetentionOrLegalHold(file.getRef()));
+            assertFalse(session.getDocument(file.getRef()).hasFacet(RetentionConstants.RECORD_FACET));
+        }
+    }
+
+
+    @Test
+    public void shouldBeAuthorizedToAttachRule() throws LoginException {
+        ACP acp = new ACPImpl();
+        ACE allowAttachRule = new ACE("user", RetentionConstants.MANAGE_RECORD_PERMISSION, true);
+        ACL acl = new ACLImpl();
+        acl.setACEs(new ACE[] { allowAttachRule });
+        acp.addACL(acl);
+        file.setACP(acp, true);
+        file = session.saveDocument(file);
+        try (CloseableCoreSession userSession = CoreInstance.openCoreSession(session.getRepositoryName(), "user")) {
+            file = service.attachRule(file, createManualImmediateRuleMillis(5000L), userSession);
+            assertTrue(userSession.isUnderRetentionOrLegalHold(file.getRef()));
+        }
+    }
+
+    @Test
+    public void shouldBeAuthorizedToSetLegalHold() throws LoginException {
+        ACP acp = new ACPImpl();
+        ACE allowLegalHold = new ACE("user", RetentionConstants.MANAGE_LEGAL_HOLD_PERMISSION, true);
+        ACL acl = new ACLImpl();
+        acl.setACEs(new ACE[] { allowLegalHold });
+        acp.addACL(acl);
+        file.setACP(acp, true);
+        file = session.saveDocument(file);
```

Comment:
can be replaced by the use of `acp.getOrCreateACL()...`

```
        ACP acp = new ACPImpl();
        acp.getOrCreateACL().add(new ACE("user", RetentionConstants.MANAGE_RECORD_PERMISSION, true))
        file.setACP(acp, true);
        file = session.saveDocument(file);
```

Reply:
Leaving for later cleanup in the numerous tests having similar code.